eqcorrscan/doc/submodules/utils.correlate.rst

```diff
@@ -17,5 +17,10 @@ correlate
        multichannel_normxcorr
        numpy_normxcorr
        time_multi_normxcorr
+       get_array_xcorr
+       get_stream_xcorr
+       register_array_xcorr
+       stream_norm_xcorr
+
```

Comment:
I wonder if this file would be a useful place to introduce the notion of swappable correlation functions, and a little example? It would also be good to have a short example of using match_filter with a different correlation function.

Reply:
Although that example should go in the match_filter docstring, I will comment there also.

Reply:
Ok, I may need a bit of help with this one. In past projects I have always used jupyter notebooks as the main form of documentation, how would I add the example here? Just with markdown syntax after the sphinx stuff?

Reply:
Yup, markdown syntax is the way to do - you should check how it renders though by making the docs:
From within the EQcorrscan/eqcorrscan/docs directory run `make html`, then open the EQcorrscan/eqcorrscan/docs/html/index.html file in the browser and navigate to the page you want just to check that it looks like useful Python.

Reply:
Ok, just out of curiosity is there a way to run the python in the markdown to make sure the docs stay fresh? To me, the appeal of jupyter notebooks is they can be run (even as part of the tests or separate CI pipe) to make sure nothing is broken before they are rendered to html.

Reply:
As long as you have:
```python
if __name__ == '__main__':
    import doctest
    doctest.main()
```
at the bottom of the file (which most/all files should have that are not solely test files themselves), then purest will pick up the tests every time it runs, and you can run the file using pytest or python to run the doctests. Doctests will find and run all examples in the docs.

Reply:
This applies for the .rst files in the doc directory as well?

Reply:
No, those aren't looked in. How I get those to run is a little convoluted, but I have done it for the tutorials:
- write a script and put it in the tutorials directory
- include that file in the .rst (with a line: `.. literalinclude:: ../../tutorials/<test_script>.py` in the rst)
- call that file from the tests/tutorial_tests.py file to run a test on it.

Reply:
Although, it looks like I could do it: https://docs.pytest.org/en/latest/doctest.html

Reply:
Note to self here: #142 
If you want to write your examples in rst let me know and I can prioritise getting that running in the next couple of days so you can rebase and run things.